### PR TITLE
DOCSP-42305: Index page + repo cleanup

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-scala/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42305>
Staging - <https://deploy-preview-56--docs-scala.netlify.app/>

## Self-Review Checklist

- [ x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -28,53 +22,91 @@
...
+.. Compatibility
+.. ------------
+
+.. To learn about the versions of the {+mdb-server+} and the C language that are compatible with 
+.. each version of the {+driver-short+}, see the :ref:`Compatibility <scala-compatibility>` section.
+
+Learn
+------
+
+Visit the Developer Hub to learn more about the {+driver-long+}.
```
Comment (jordan-smith721) [inline source/index.txt:103]: Are we anticipating adding more subheadings under this learn section? If not we can probably just stick with the Dev Hub section nested under this

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ one question/maybe suggestion

File: source/index.txt (hunk 1 above)
Comment (mcmorisi) [reply source/index.txt:103]: I don't think there's any scoped additions to this learn section. Do you mean removing the top-level Learn section and pulling the Developer Hub section up to the top level of the page's ToC?

Comment (jordan-smith721) [reply source/index.txt]: Yeah that's what I was thinking, but happy to differ that decision to later when this part is worked on

**Pull request merged**